src/sock.c

```diff
@@ -99,6 +100,17 @@ sock_t sock_connect(const char * const host, const unsigned int port)
         if (sock < 0)
             continue;
 
+	/* turn on keepalive */
```

Comment:
Why do you want to turn keepalives? Is there any issue around this? Libstrophe supports windows and posix systems, so such low-level code should be added carefully. For example, as I googled, windows doesn't support last 3 TCP options and thus can return error.